BookShelf/Controllers/BookController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using BookShelf.Models;
+using BookShelf.BookService;
+
+namespace BookShelf.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BookController : ControllerBase
+    {
+        private IBook _books;
+
+        public BookController(IBook booklist)
+        {
+            _books = booklist;
+        }
+
+        [HttpGet]
+        public ActionResult<List<Book>> GetAll()
+        {
+            return _books.getAll();
+        }
+
+        [HttpGet("{id}", Name = "GetBook")]
```

Comment:
You don't need to specify route GetBook for Get. By all conventions, it`s good to have such url host/books/1